From 13aadf79659dd238b618c8be7c1de44960bd5d50 Mon Sep 17 00:00:00 2001
From: Romain Naour <romain.naour@gmail.com>
Date: Fri, 11 Nov 2016 22:20:03 +0100
Subject: [PATCH] sane_backend: add missing config.h

We should include config.h from sanei_backend.h in order to use the
correct if/else HAVE_FOO.

For some reason with Glibc or uClibc there is no problem but with musl
we have the following weird issue:

In file included from epsonds.h:41:0,
                 from epsonds-jpeg.c:18:
../include/sane/sanei_backend.h:99:33: error: expected ';', identifier or '(' before 'int'
 # define sigset_t               int
                                 ^
../include/sane/sanei_backend.h:99:33: warning: useless type name in empty declaration

That's because HAVE_SIGPROCMASK is not defined although it's correctly
detected by the configure script.

$ grep config.log
config.log:#define HAVE_SIGPROCMASK 1

So, include config.h to avoid to redefine sigset_t.

Fixes:
http://autobuild.buildroot.net/results/9f1/9f1f1cb727b5c5407e69172280a3dee880e55cdf

Signed-off-by: Romain Naour <romain.naour@gmail.com>
---
 include/sane/sanei_backend.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/sane/sanei_backend.h b/include/sane/sanei_backend.h
index 1b5afe2..2a96532 100644
--- a/include/sane/sanei_backend.h
+++ b/include/sane/sanei_backend.h
@@ -8,6 +8,7 @@
  * @sa sanei.h sanei_thread.h
  */
 
+#include "../include/sane/config.h"
 
 /*
  * Compiler related options
-- 
2.5.5

